app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -189,6 +234,38 @@ class ParkingViewModel(
     parkingRepository.updateParking(parking, {}, {})
   }
 
+  /**
+   * Updates the list of closest parkings.
+   *
+   * @param nbRequestLeft: number of tiles left to fetch the parkings from. If nbRequestLeft is 0,
+   *   the function will update the closest parkings and if the result is empty, it will increment
+   *   the radius.
+   */
+  private fun updateClosestParkings(nbRequestLeft: Int) {
+    if (_circleCenter.value == null || nbRequestLeft != 0) return // avoid updating if not ready
+    _closestParkings.value =
+        _rectParkings.value
+            .filter { parking ->
+              TurfMeasurement.distance(_circleCenter.value!!, parking.location.center) *
+                  KM_TO_METERS <= _radius.value
+            }
+            .sortedBy { parking ->
+              TurfMeasurement.distance(_circleCenter.value!!, parking.location.center)
+            }
+            .filter { parking ->
+              (_selectedProtection.value.isEmpty() ||
+                  _selectedProtection.value.contains(parking.protection)) &&
+                  (selectedRackTypes.value.isEmpty() ||
+                      _selectedRackTypes.value.contains(parking.rackType)) &&
+                  (_selectedCapacities.value.isEmpty() ||
+                      _selectedCapacities.value.contains(parking.capacity)) &&
+                  (!_onlyWithCCTV.value || parking.hasSecurity)
+            }
+    if (_closestParkings.value.size < 5 || _radius.value == MAX_RADIUS) {
```

Comment:
5 here seems a bit arbitrary. Why 5?

Reply:
You're right I should use Magic number. 
If the UI is not entierly filled with items,  the code to fetch new one when the user arrives at the end of the list  is never read. I've put 5 randomly, but the correct number should be equal to screenheight/itemheight